var/httpd/htdocs/js/Core.Agent.DynamicFieldDBSearch.js

```diff
@@ -605,8 +605,9 @@ Core.Agent.DynamicFieldDBSearch = (function(TargetNS) {
         }
 
         var FieldNameLong = Field;
-        if ( ActivityDialogID != '' ) {
-            Field = Field.substr(0, Field.indexOf('_' + ActivityDialogID));
+        var IndexOfActivityDialogID = Field.substr(0, Field.indexOf('_' + ActivityDialogID));
+        if ( ActivityDialogID != '' && IndexOfActivityDialogID > 0 ) {
```

Comment:
I would suggest to shorten it a bit, e.g. like:

if ( ActivityDialogID != '' && Field.indexOf('_' + ActivityDialogID > 0) ) {

However, this assumes that the original DynamicFieldName has a length > 0, but as far as I know, a DynamicField is required to have a name.

Reply:
I used an extra variable as the result is used twice (see new line 610).

I don't get the last pragraph....

Reply:
The FieldName here is '[DynamicFieldName]_[ActivityDialogID]', and if the DynamicFieldName would be empty, 0 would be a valid index for '_' + ActivityDialogID. But I can't think of a case where this would happen, so it looks ok to me. @svenoe What do you think?